src/puyapy_mocks/op.py

```diff
@@ -0,0 +1,67 @@
+from puyapy_mocks._ctx_state import active_ctx
+from puyapy_mocks._primatives import (
+    Bytes,
+    UInt64,
+)
+from puyapy_mocks._reference import (
+    Account,
+)
+
+
+def btoi(a: Bytes | bytes, /) -> UInt64:
+    if isinstance(a, Bytes):
+        a = bytes(a)
+    return UInt64(int.from_bytes(a))
+
+
+def itob(a: int | UInt64, /) -> Bytes:
+    if isinstance(a, UInt64):
+        a = int(a)
+    return Bytes(int.to_bytes(a, 8))
+
+
+class Global:
+    pass
+
+
+# TODO: NC - Come back to this and see if we can simplify
+class classproperty(property):  # noqa: N801
+    pass
+
+
+class StaticProperty(type):
+    def __new__(cls, name, bases, props):  # type: ignore[no-untyped-def]  # noqa: ANN204, ANN001
+        class_properties = {}
+        to_remove = {}
+        for key, value in props.items():
+            if isinstance(value, (classproperty)):
+                class_properties[key] = value
+                if isinstance(value, classproperty):
+                    to_remove[key] = value
+
+        for key in to_remove:
+            props.pop(key)
+
+        hoist_meta = type("HoistMeta", (type,), class_properties)
+        return hoist_meta(name, bases, props)
+
+
+class Txn(metaclass=StaticProperty):
+    @classmethod
+    def application_args(cls, index: int) -> bytes:
+        return active_ctx().args[index]
+
+    @classproperty
+    def num_app_args(self) -> int:
+        return len(active_ctx().args)
+
+
+class AssetHoldingGet:
+    @staticmethod
+    def asset_balance(a: Account | UInt64 | int, b: UInt64 | int, /) -> tuple[UInt64, bool]:
+        if isinstance(a, Account):
+            address = a.address
+        elif isinstance(a, (int, UInt64)):
+            address = str(a)
+
+        return UInt64(active_ctx().get_balance(address, int(b))), False
```

Comment:
@neilcampbell what happens when a is of type UInt64? Wouldn't address be None in this case ?

Reply:
Yeah this code is very wrong, so we'll need to handle int/UInt64 address values properly.